src/Lean/Widget/InteractiveGoal.lean

```diff
@@ -136,6 +137,8 @@ def goalToInteractive (mvarId : MVarId) : MetaM InteractiveGoal := do
     let userName? := match mvarDecl.userName with
       | Name.anonymous => none
       | name           => some <| toString name.eraseMacroScopes
-    return { hyps, type := goalFmt, userName? }
+    let mut pref := "⊢"
+    if isLHSGoal? mvarDecl.type |>.isSome then pref := "|"
```

Comment:
It would be interesting to make this choice extensible in the future -- for example, for Iris, we may want to hide it completely so that we can append our own contexts (/cc @larsk21). But I am fine with this implementation for this PR.

Reply:
Given the above, however, it would be good to include the trailing space in `goalPrefix` so that we don't have to change the client again when we allow arbitrary and potentially empty prefixes.